src/gameplay/glicko2.ts

```diff
@@ -0,0 +1,136 @@
+// @ts-nocheck
+
+const computeG = (phi) => {
+  return 1 / Math.sqrt(1 + (3 * phi ** 2) / Math.PI ** 2);
+};
+
+const computeE = (mu, mu_j, phi_j) => {
+  return 1 / (1 + Math.exp(-1 * computeG(phi_j) * (mu - mu_j)));
+};
+
+
+const computeNewVolatility = (delta, phi, v, sigma, tau) => {
+  const epsilon = 0.000001;
+  const a = Math.log(sigma ** 2);
+  const computeF = (x) => {
+    let output = 0;
+
+    output += Math.E ** x * (delta ** 2 - phi ** 2 - v - Math.E ** x) / (2 * (phi ** 2 + v + Math.E ** x) ** 2);
+    output -= (x - a) / tau ** 2;
+    return output;
+  };
+  
+  
+  let A = a;
+  let B = 0;
+  if (delta ** 2 > phi ** 2 + v) {
+    B = Math.log(delta ** 2 - phi ** 2 - v);
+  } else {
+    let k = 1;
+    while (computeF(a - k * tau) < 0) {
+      k += 1;
+    }
+    B = a - k * tau;
+  }
+
+
+  let f_A = computeF(A);
+  let f_B = computeF(B);
+
+  
+  while (Math.abs(B - A) > epsilon) {
+    const C = A + (A - B) * f_A / (f_B - f_A);
+    const f_C = computeF(C);
+    if (f_C * f_B <= 0) {
+      A = B;
+      f_A = f_B;
+    } else {
+      f_A /= 2;
+    }
+
+    B = C;
+    f_B = f_C;
+  }
+
+  return Math.E ** (A / 2);
+};
+
+export const glicko2Update = (playerData, matchHistory) => {
```

Comment:
Let's make this stuff into a class :). You can keep them as static methods, but it'll help clarify what the public and private methods are so it's easier for future readers!